sql/analyzer/resolve_columns.go

```diff
@@ -770,6 +772,19 @@ func indexColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (map[
 		}
 	}
 
+	// having node need access to all table references within its children node,
+	// so reset idx for any additional schema
+	switch n.(type) {
+	case *plan.Having:
+
+		transform.Inspect(n, func(node sql.Node) bool {
+			for _, child := range n.Children() {
+				indexChildNode(child)
+			}
+			return true
+		})
+	}
+
```

Comment:
I am still skeptical about this code... it seems odd that we'd handle `having` differently than other nodes. 

I just tried commenting out this block and lines 765 and 766, and all the tests still executed correctly for me. Are you sure we really need this?